tests/Integration/Behaviour/Features/Context/ShopFeatureContext.php

```diff
@@ -65,16 +73,104 @@ public function shopWithNameExists(string $reference, string $shopName)
     {
         Shop::setContext(Shop::CONTEXT_SHOP, Configuration::get('PS_SHOP_DEFAULT'));
     }
 
     /**
      * @Given multiple shop context is loaded
      */
-    public function multipleShopContextIsLoaded()
+    public function multipleShopContextIsLoaded(): void
     {
         Shop::setContext(Shop::CONTEXT_ALL);
     }
+
+    /**
+     * @Given I add a shop :reference with name :shopName for the group :shopGroupName
+     *
+     * @param string $reference
+     * @param string $shopName
+     * @param string $shopGroupName
+     */
+    public function addShop(string $reference, string $shopName, string $shopGroupName): void
+    {
+        $shop = new Shop();
+        $shop->active = true;
+        $shop->id_shop_group = ShopGroup::getIdByName($shopGroupName);
+        $shop->id_category = 2;
+        $shop->theme_name = _THEME_NAME_;
+        $shop->name = $shopName;
+        if (!$shop->add()) {
+            throw new RuntimeException(sprintf('Could not create shop: %s', Db::getInstance()->getMsgError()));
+        }
+        $shop->setTheme();
+
+        SharedStorage::getStorage()->set($reference, $shop);
+    }
+
+    /**
+     * @When I search for shops with the term :searchTerm I should get the following results:
+     *
+     * @param string $searchTerm
+     * @param TableNode $table
+     */
+    public function assertFoundShops(string $searchTerm, TableNode $table): void
+    {
+        $expectedShops = $table->getColumnsHash();
+        $foundShops = $this->getQueryBus()->handle(new SearchShops($searchTerm));
+
+        foreach ($expectedShops as $key => $currentExpectedShop) {
+            $wasCurrentExpectedShopFound = false;
+            foreach ($foundShops['shops'] as $currentFoundShop) {
+                if (strtolower($currentExpectedShop['name']) == strtolower($currentFoundShop['name'])) {
+                    $wasCurrentExpectedShopFound = true;
+                    Assert::assertEquals(
+                        $currentExpectedShop['group_name'],
+                        $currentFoundShop['group_name'],
+                        sprintf(
+                            'Expected and found shops\'s groups don\'t match (%s and %s)',
+                            $currentExpectedShop['group_name'],
+                            $currentFoundShop['group_name']
+                        )
+                    );
+                    continue;
+                }
+            }
+            if (!$wasCurrentExpectedShopFound) {
+                throw new RuntimeException(sprintf(
+                    'Expected shop with name %s in shop group %s was not found',
+                    $currentExpectedShop['name'],
+                    $currentExpectedShop['group_name']
+                ));
+            }
+        }
+    }
+
+    /**
+     * Set the list of found shops in the last research
+     *
+     * @param array $shops
+     */
+    private function setLastSearchShopsResult(array $shops): void
```

Comment:
This function is now useless It's a good thing, storing the result of a step to test it later is a bit dangerous as the whole test becomes stateful